TensorLib/Basic.lean

```diff
@@ -2,4 +2,98 @@ namespace TensorLib
 
```

Comment:
I like to reserve the Basic.lean file to just import all the other relevant files in the directory. You may want to create two files, one for the tensor elements, and one for the tensor themselve.